Makefile

```diff
@@ -0,0 +1,140 @@
+VERSION = $(shell grep "^Version: " DESCRIPTION | cut -f2 -d" ")
+MATLAB_PACKAGE = octsympy
+BUILD_DIR = tmp
+MATLAB_PKG_DIR=${MATLAB_PACKAGE}-matlab-${VERSION}
+OCTAVE_RELEASE_TARBALL = ${BUILD_DIR}/${PACKAGE}-${VERSION}.tar
+OCTAVE_RELEASE_TARBALL_COMPRESSED = ${OCTAVE_RELEASE_TARBALL}.gz
+INSTALLED_PACKAGE = ~/octave/${PACKAGE}-${VERSION}/packinfo/DESCRIPTION
+HTML_DIR = ${BUILD_DIR}/${PACKAGE}-html
+HTML_TARBALL_COMPRESSED = ${HTML_DIR}.tar.gz
+
+OCTAVE ?= octave
+MATLAB ?= matlab
+
+.PHONY: help clean install test doctest pkg html matlab_test matlab_pkg
+
+help:
+	@echo Available rules:
+	@echo "  clean              clean all temporary files"
+	@echo "  install            install package in Octave"
+	@echo "  test               run tests with Octave"
+	@echo "  doctest            run doctests with Octave"
+	@echo "  pkg                create Octave package (${OCTAVE_RELEASE_TARBALL_COMPRESSED})"
+	@echo "  html               create Octave Forge website"
+	@echo
+	@echo "  matlab_test        run tests with Matlab"
+	@echo "  matlab_pkg         create Matlab package (${MATLAB_PKG_DIR}.zip)"
+
+
+
+${BUILD_DIR} ${BUILD_DIR}/${MATLAB_PKG_DIR}/private ${BUILD_DIR}/${MATLAB_PKG_DIR}/tests_matlab ${BUILD_DIR}/${MATLAB_PKG_DIR}/@sym ${BUILD_DIR}/${MATLAB_PKG_DIR}/@symfun ${BUILD_DIR}/${MATLAB_PKG_DIR}/@logical:
+	mkdir -p "$@"
+
+clean:
+	rm -rf "${BUILD_DIR}"
+	@#rm -f fntests.log
+	rm -f octsympy_tests.log
+
+test:
+	@echo "Testing package in GNU Octave ..."
+	@$(OCTAVE) --no-gui --silent --path "${PWD}/inst" \
+		--eval "set (0, 'defaultfigurevisible', 'off'); \
+		 octsympy_tests; \
+		 sympref reset"
+	@! grep '!!!!! test failed' octsympy_tests.log
+	@echo
+
+doctest:
+	@# Workaround for OctSymPy issue 273, we must pre-initialize the package
+	@# Otherwise, it will make the doctests fail
+	@echo "Testing documentation strings ..."
+	@$(OCTAVE) --no-gui --silent --path "${PWD}/inst" \
+		--eval "pkg load doctest; \
+		 sym ('x'); \
+		 set (0, 'defaultfigurevisible', 'off'); \
+		 success = doctest('inst/'); \
+		 sympref reset; \
+		 exit (!success)"
+	@echo
+
+
+## Install in Octave (locally)
+install: ${INSTALLED_PACKAGE}
+${INSTALLED_PACKAGE}: ${OCTAVE_RELEASE_TARBALL_COMPRESSED}
+	$(OCTAVE) --silent --eval "pkg install $<"
+
+## Package release for Octave
+${OCTAVE_RELEASE_TARBALL}: .git/index | ${BUILD_DIR}
+	git archive --output="$@" --prefix=${PACKAGE}-${VERSION}/ HEAD
+	tar --delete --file "$@" ${PACKAGE}-${VERSION}/README.matlab.md
+	tar --delete --file "$@" ${PACKAGE}-${VERSION}/HOWTO-release.md
+	tar --delete --file "$@" ${PACKAGE}-${VERSION}/README.bundled.md
+	tar --delete --file "$@" ${PACKAGE}-${VERSION}/TODO.md
+	tar --delete --file "$@" ${PACKAGE}-${VERSION}/testing
+	tar --delete --file "$@" ${PACKAGE}-${VERSION}/util
+	tar --delete --file "$@" ${PACKAGE}-${VERSION}/.gitignore
+	tar --delete --file "$@" ${PACKAGE}-${VERSION}/.mailmap
+	tar --delete --file "$@" ${PACKAGE}-${VERSION}/screenshot*.png
```

Comment:
Or you could use `git archive` and immediately unpack it into a temporary directory, fix up the directory with `chmod` and `rm`, and then tar and compress the result. That's probably what I would do, I try to avoid GNU options like `--delete`.